package/src/prmon.cpp

```diff
@@ -173,10 +173,56 @@ int MemoryMonitor(const pid_t mpid, const std::string filename, const std::strin
      Value& v1 = d["Max"];
      Value& v2 = d["Avg"];
 
-     startTime = time(0);
+     // Get mother process start time
+     char stat_buffer[64], sbuffer[2048], *tsbuffer;
+
+     snprintf(stat_buffer,64,"/proc/%llu/stat",(unsigned long long)mpid);
+
+     FILE *stat_file = fopen(stat_buffer,"r");
+     unsigned long long thispid(0);
+     unsigned long long starttime(0);
+
+     if(stat_file==0) {
+       std::cerr << "Cannot open stat buffer for mother pid to get the starttime" << std::endl; 
+     }
+     else {
+       while(fgets(sbuffer,2048,stat_file)) {
+         if(sscanf(sbuffer, "%80llu", &thispid)) {
+           if(thispid!=(unsigned long long)mpid) continue;
+           tsbuffer = strchr (sbuffer, ')');
+           if(sscanf(tsbuffer + 2 , "%*c %*d %*d %*d %*d %*d %*u %*u %*u %*u %*u %*u %*u %*d %*d %*d %*d %*d %*d %80llu", &starttime)) {
+             break;
+           }
+         }
+       }
+       fclose(stat_file);
+     }
+
+     // Get system uptime
+     snprintf(stat_buffer,64,"/proc/uptime");
+
+     fopen(stat_buffer,"r");
```

Comment:
Bug! Should be

`stat_file = fopen(stat_buffer,"r");`

Reply:
Wow! I'm a bit surprised how it actually worked fine w/ this bug. Should be fixed now.